commands/build.js

```diff
@@ -80,9 +103,122 @@ function buildLumin (argv) {
 }
 
 module.exports = argv => {
-  npmInstallIfNeeded(() => {
+  npmInstallIfNeeded(`${process.cwd()}`, () => {
     if (argv.target === 'lumin') {
-      buildLumin(argv);
+      if (isComponents()) {
+        npmInstallIfNeeded(`${process.cwd()}/lumin`, () => {
+          buildLuminComponents(argv);
+        });
+      } else {
+        buildLumin(argv, "#!/system/bin/script/mxs\nimport './src/main.js';\n");
+      }
+    } else if (argv.target === 'android') {
+      npmInstallIfNeeded(`${process.cwd()}/reactnative`, () => {
+        buildAndroid(argv);
+      });
+    } else if (argv.target === 'ios') {
+      npmInstallIfNeeded(`${process.cwd()}/reactnative`, () => {
+        buildiOS(argv);
+      });
+    } else {
+      console.error('The target must be either lumin, ios or android!');
     }
   });
 };
+
+function isComponents () {
```

Comment:
This function checks if we have this 'monorepo' structure:
-common
-lumin
-reactnative
-src

Or legacy Components/Vanilla MagicScript project. If there is a rollup.config.js file in 'lumin' directory, we are dealing with the 'monorepo' components structure and should navigate to 'lumin' directory to execute commands. Otherwise we are dealing with legacy projects and we should execute commands in the current directory

Reply:
so the name isn't really a good function name, can you please updated it to something more descriptive and relevant to what it is actually checking.

Reply:
Sure, I changed it to isMultiplatformStructure()